android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/RibEvents.kt

```diff
@@ -16,28 +16,82 @@
 package com.uber.rib.core
 
 import io.reactivex.Observable
+import kotlin.reflect.KClass
 import kotlinx.coroutines.channels.BufferOverflow
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.rx2.asObservable
 
-public class RibEvents private constructor() {
+public object RibEvents {
 
-  private val _events = MutableSharedFlow<RibEvent>(0, 1, BufferOverflow.DROP_OLDEST)
-  public val events: Observable<RibEvent> = _events.asObservable()
+  private val mutableRouterEvents =
+    MutableSharedFlow<RibRouterEvent>(0, 1, BufferOverflow.DROP_OLDEST)
+  private val mutableRibDurationEvents =
+    MutableSharedFlow<RibEventDurationData>(0, 1, BufferOverflow.DROP_OLDEST)
+
+  @JvmStatic
+  public val routerEvents: Observable<RibRouterEvent>
+    get() = mutableRouterEvents.asObservable()
+
+  @JvmStatic
+  public val ribDurationEvents: Observable<RibEventDurationData>
+    get() = mutableRibDurationEvents.asObservable()
 
   /**
    * @param eventType [RibEventType]
    * @param child [Router]
    * @param parent [Router] and null for the root ribs that are directly attached to
    *   RibActivity/Fragment
    */
-  public fun emitEvent(eventType: RibEventType, child: Router<*>, parent: Router<*>?) {
-    _events.tryEmit(RibEvent(eventType, child, parent))
+  public fun emitRouterEvent(eventType: RibEventType, child: Router<*>, parent: Router<*>?) {
+    mutableRouterEvents.tryEmit(RibRouterEvent(eventType, child, parent))
   }
 
-  public companion object {
-    private val instance: RibEvents = RibEvents()
+  /**
+   * Emits emission of ATTACHED/DETACHED events for each RIB component.
+   *
+   * @param ribClass Class names for custom RIB implementations (e.g. LoggedInInteractor,
+   *   UiRibWorker, etc)
+   * @param ribComponentType The RIB component type (e.g. Interactor, Router, Presenter)
+   * @param ribEventType RIB event type (e.g. ATTACH/DETACH)
+   * @param totalBindingDurationMilli Total duration (in ms) of each ATTACH/DETACH events
+   */
+  internal fun emitRibEventDuration(
+    ribClass: KClass<*>,
+    ribComponentType: RibComponentType,
+    ribEventType: RibEventType,
+    totalBindingDurationMilli: Long,
+  ) {
+    val ribClassName = ribClass.qualifiedName
 
-    @JvmStatic public fun getInstance(): RibEvents = instance
```

Comment:
@JavaOnly annotation to keep singleton

Reply:
Actually, decided to remove this all-together and to replace existing usages moving forward